File: 100-Days-Series/Day 34/InsertionSort.py
# Function for Insertion Sort
def getInsertionSort(arr):
    for i in range(1,len(arr),1):
        key=arr[i]
        j=i-1
        while j>=0 and arr[j]>key:
            arr[j+1]=arr[j]
            j=j-1
        arr[j+1]=key
    return arr

# Let's Start here
arr=[3,2,5,6,7,1]
print('Before Sorting...')
print(arr)
print('After Insertion Sorting...')
print(getInsertionSort(arr))

File: 100-Days-Series/Day 34/InsertionSort.js
// Function for Insertion Sort
const getInsertionSort=(arr)=>{
    for(let i=1;i<arr.length;i++){
        let key=arr[i];
        let j=i-1;
        while(j>=0 && arr[j]>key){
            arr[j+1]=arr[j];
            j--;
        }
        arr[j+1]=key;
    }
}
// Let's Start here
arr=new Array(3,4,5,1,7,6,8,9);
console.log("Before Sorting...");
console.log(arr);
console.log("After Sorting...");
getInsertionSort(arr);
console.log(arr);